app/app/explore/components/AddLiquidityModal.tsx

```diff
@@ -0,0 +1,274 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import Image from "next/image";
+import Button from "@/app/button";
+import Modal from "@/app/modal";
+import { Balance, getBalance, PoolDetail as IPoolDetail } from "@/midgard";
+import { Slider } from "@shared/components/ui";
+import { twMerge } from "tailwind-merge";
+import {
+  getAssetShortSymbol,
+  getLogoPath,
+  isERC20,
+  normalizeAddress,
+} from "@/app/utils";
+import { useAppState } from "@/utils/context";
+import { useLiquidityPosition } from "@/hooks/useLiquidityPosition";
+import ErrorCard from "@/app/errorCard";
+import { useContracts } from "@/hooks/useContracts";
+
+interface AddLiquidityModalProps {
+  pool: IPoolDetail;
+  runePriceUSD: number;
+  onClose: () => void;
+}
+
+export default function AddLiquidityModal({
+  pool,
+  runePriceUSD,
+  onClose,
+}: AddLiquidityModalProps) {
+  const { wallet } = useAppState();
+  const { loading, error, addLiquidity } = useLiquidityPosition({ pool });
+  const [selectedTab] = useState("single");
```

Comment:
"single" and "double" should be enums

Reply:
btw we should only allow executing the dlp flow if the user has connected a wallet that supports native rune (ie CTRL now, Vultisig later).